profiles/seccomp/seccomp_linux.go

```diff
@@ -29,16 +27,42 @@ func LoadProfile(body string, rs *specs.Spec) (*specs.LinuxSeccomp, error) {
 	return setupSeccomp(&config, rs)
 }
 
+// libseccomp string => seccomp arch
 var nativeToSeccomp = map[string]types.Arch{
+	"x86":         types.ArchX86,
```

Comment:
couldn't we change this to `goarchToSeccomp` ?

so change this to "386" ?

Reply:
ah, nevermind; looks like `arch` is used further down (didn't show in the diff)

Reply:
Guess we could still do it, but we would need both mapping (but it would change to GOARCH -> native, and GOARCH -> seccomp)

Reply:
Yeah this was pretty much why it was failing. Didn't realize we were using the libseccomp arch string.

Reply:
Do you think it will be cleaner to have `GOARCH -> nativeArch` and `GOARCh -> seccompArch`, so that they can be used independently?

```patch
diff --git a/profiles/seccomp/seccomp_linux.go b/profiles/seccomp/seccomp_linux.go
index 7847085a0a..ac6441d6c3 100644
--- a/profiles/seccomp/seccomp_linux.go
+++ b/profiles/seccomp/seccomp_linux.go
@@ -27,16 +27,16 @@ func LoadProfile(body string, rs *specs.Spec) (*specs.LinuxSeccomp, error) {
        return setupSeccomp(&config, rs)
 }

-// libseccomp string => seccomp arch
-var nativeToSeccomp = map[string]types.Arch{
-       "x86":         types.ArchX86,
+// GOARCH => seccomp arch
+var goToSeccomp = map[string]types.Arch{
+       "386":         types.ArchX86,
        "amd64":       types.ArchX86_64,
        "arm":         types.ArchARM,
        "arm64":       types.ArchAARCH64,
        "mips64":      types.ArchMIPS64,
-       "mips64n32":   types.ArchMIPS64N32,
-       "mipsel64":    types.ArchMIPSEL64,
-       "mips3l64n32": types.ArchMIPSEL64N32,
+       "mips64p32":   types.ArchMIPS64N32,
+       "mips64le":    types.ArchMIPSEL64,
+       "mips64p32le": types.ArchMIPSEL64N32,
        "mipsle":      types.ArchMIPSEL,
        "ppc":         types.ArchPPC,
        "ppc64":       types.ArchPPC64,
@@ -97,8 +97,7 @@ func setupSeccomp(config *types.Seccomp, rs *specs.Spec) (*specs.LinuxSeccomp, e
                }
        }

-       arch := goToNative[runtime.GOARCH]
-       seccompArch, archExists := nativeToSeccomp[arch]
+       seccompArch, archExists := goToSeccomp[runtime.GOARCH]

        if len(config.ArchMap) != 0 && archExists {
                for _, a := range config.ArchMap {
@@ -115,6 +114,8 @@ func setupSeccomp(config *types.Seccomp, rs *specs.Spec) (*specs.LinuxSeccomp, e
        newConfig.DefaultAction = specs.LinuxSeccompAction(config.DefaultAction)

 Loop:
+       arch := goToNative[runtime.GOARCH]
+
        // Loop through all syscall blocks and convert them to libcontainer format after filtering them
        for _, call := range config.Syscalls {
                if len(call.Excludes.Arches) > 0 {
```

Reply:
It's different, I don't think it's better or worse.

Reply:
Yeah, it's not a blocker, as non of this is exported. Thought it would be slightly easier to understand, as we wouldn't be mapping twice (GOOS -> native -> seccomp).

I can open a follow-up to discuss